src/mem_table.rs

```diff
@@ -135,7 +135,7 @@ impl WritableKVTable {
 }
 
 impl KVTable {
-    fn new() -> Self {
+    pub fn new() -> Self {
```

Comment:
This feels far too permissive. I think `pub(crate)` at most. But I would prefer not to expose it at all if possible...